Modules/Filtering/ImageGrid/include/itkResampleImageFilter.hxx

```diff
@@ -587,8 +587,10 @@ ResampleImageFilter<TInputImage, TOutputImage, TInterpolatorPrecisionType, TTran
   GenerateOutputInformation()
 {
   // Call the superclass' implementation of this method
-  Superclass::GenerateOutputInformation();
-
+  if (InputImageDimension == OutputImageDimension)
```

Comment:
Why should we skip this? `ImageToImageFilter` supports different dimensionality of input and output.

Reply:
It should! but it throws the following exception [here](https://github.com/InsightSoftwareConsortium/ITK/blob/2074c2f9e5ed3f087d0e2059f8e0e8992fcad7ef/Modules/Core/Common/include/itkImageBase.hxx#L308). 
"itk::ERROR: Image: itk::ImageBase::CopyInformation() cannot cast PKN3itk10DataObjectE to PKN3itk9ImageBaseILj3EEE"

Reply:
The documentation of `GenerateOutputInformation` explains it:
https://github.com/InsightSoftwareConsortium/ITK/blob/0d1aaefef9a4ccdd53569dc1bc1f07f46209b4bb/Modules/Core/Common/include/itkProcessObject.h#L374-L390
Indeed, we cannot just copy the information as in default implementation
https://github.com/InsightSoftwareConsortium/ITK/blob/0d1aaefef9a4ccdd53569dc1bc1f07f46209b4bb/Modules/Core/Common/src/itkProcessObject.cxx#L1789-L1795
and I am not sure that we need to.